### ISSUE TITLE
How to model a mixed effects model with three (or more) levels of nested random effects

### ISSUE THREAD

**Issue by hsuominen** [statsmodels/statsmodels#6802]
Similarly to #3500, I would like to know if it is possible to fit a mixed random effects
model with three random effects when the factors are nested within each other? The extension from the 2 to higher levels of nesting is not obvious to me from the documentation/examples

E.g. in R this would be achieved as follows:
`lmer(y ~ (1|group1/group2/group3), data=df)`

**Comment by kshedden** [comment]
Yes, it is possible.  We have an example at the link below that has two levels of grouping:

https://github.com/statsmodels/statsmodels/blob/master/examples/notebooks/variance_components.ipynb

For a three level analysis, you would use keyword arguments like this:

```
re_formula="1", groups="group1", vc_formula={"group2": "0 + C(group2)", "group3": "0 + C(group3)"}
```

Here, group1, group2, group3 are treated as categorical variables.   The codes of each deeper level should be unique across the shallower levels, something like this:

```
group1   group2   group3
1        1        1        
1        1        1
1        1        2
1        1        2
1        2        3
1        2        3
1        2        4
1        2        4
2        3        5
2        3        5
2        3        6
2        3        6
...
```

If the codes are recyled (i.e. group3=1 within group2=1 is a different group from group3=1 within group2=2), then  you should either relabel them, or perhaps using interactions would work like below, but I would want to test this to be sure:

```
re_formula="1", groups="group1", vc_formula={"group2": "0 + C(group2)", "group3": "0 + C(group3*group2)"}
```

It is very important to have the "0 + " in the formulas.

I have a PR that is aiming to improve performance for this type of analysis.  If your data set is large, it might benefit from this.  I will try to get it merged in today.  You would need to set `use_spare=True` to benefit from this.  It still will not be as fast as R, because we do not have access to a sparse Cholesky solver with the right license.

**Comment by hsuominen** [comment]
Thanks for the quick response @kshedden! Indeed I had seen that example tried this (should have specified previously):
`re_formula="1", groups="group1", vc_formula={"group2": "0 + C(group2)", "group3": "0 + C(group3)"}`
but did not have unique codes at depth - hence it was failing my testing. I'll try the interactions or fall back to relabeling

**Comment by dana89co** [comment]
Thanks for this explanation. It helped me a lot. However, I am not sure I understand why you use "re_formula="1"". What is the logic behind it?
